testing/xfpga/test_metrics_vector_c.cpp

```diff
@@ -133,35 +126,32 @@ TEST(metric_vector, test_metric_vector_02) {
  *             from config.h.<br>
  */
 TEST(metric_vector, test_metric_vector_03) {
+  fpga_metric_vector metric_vector;
+  uint64_t total;
+  // Init vector
+  EXPECT_EQ(FPGA_OK, fpga_vector_init(&metric_vector));
 
-	fpga_metric_vector metric_vector;
-	uint64_t  total;
-	// Init vector
-	EXPECT_EQ(FPGA_OK, fpga_vector_init(&metric_vector));
+  struct _fpga_enum_metric *fpga_enum_metric = NULL;
+  fpga_enum_metric = (struct _fpga_enum_metric *)malloc(sizeof(struct _fpga_enum_metric));
+  EXPECT_EQ(NULL, !fpga_enum_metric);
 
-	struct _fpga_enum_metric *fpga_enum_metric = NULL ;
-	fpga_enum_metric = (struct _fpga_enum_metric *)malloc(sizeof(struct _fpga_enum_metric));
-	EXPECT_EQ(NULL, !fpga_enum_metric);
+  EXPECT_EQ(FPGA_OK, fpga_vector_push(&metric_vector, fpga_enum_metric));
```

Comment:
Is fpga_enum_metric being freed?

Reply:
I believe the fpga_vector_free call on line 156 iterates and frees all vector elements.